ssh/Earthfile

```
@@ -0,0 +1,16 @@
+VERSION 0.7
+
+# COPY_KNOWN_HOSTS will append some known public keys into the $target_file
+COPY_KNOWN_HOSTS:
+    COMMAND
+    ARG target_file=~/.ssh/known_hosts
+    ARG target_dir=$(eval echo $(dirname $target_file))
+    COPY +known-hosts/known_hosts /tmp/.
+    RUN mkdir -p $target_dir && cat /tmp/known_hosts >> $(eval echo $target_file)
```

can this be simplified to:
```suggestion
    RUN mkdir -p $(dirname $target_file) && cat /tmp/known_hosts >> $target_file
```
I'm not sure I understand the requirement for the `eval echo` pattern.

---

when `target_file=~/.ssh/known_hosts`, then `mkdir -p $(dirname $target_file)` creates `./~/.ssh` instead of `~/.ssh`
so eval first interpolates `~`

I'll be happy to do it differently.

---

ah ha! I think that should be called out via a comment.

How about using a `LET` for this case:

```
VERSION --arg-scope-and-set 0.7

EXPAND:
  COMMAND
  ARG path
  # expanded_path expands the path, e.g. ~/ becomes /root/, or /home/user
  LET expanded_path="$(eval echo $path)"
  RUN echo "the expanded path is $expanded_path"

test:
  FROM alpine
  DO +EXPAND --path=~/.ssh/known_hosts
```

The LET is used instead of an ARG since we want to use `expanded_path` as a temporary variable and which will prevent the user from overriding it.